Brain Traumas

XDTM opened the pull request:
:cl: XDTM
add: Brain damage has been completely reworked!
remove: Brain damage now no longer simply makes you dumb. Although most of its effects have been shifted into a brain trauma.
add: Every time you take brain damage, there's a chance you'll suffer a brain trauma. There are many variations of brain traumas, split in mild, severe, and special.
add: Mild brain traumas are the easiest to get, can be lightly to moderately annoying, and can be cured with mannitol and time.
add: Severe brain traumas are much rarer and require extensive brain damage before you have a chance to get them; they are usually very debilitating. Unlike mild traumas, they require surgery to cure. A new surgery procedure has been added for this, the aptly named Brain Surgery. It can also heal minor traumas.
add: Special brain traumas are rarely gained in place of Severe traumas: they are either complex or beneficial.
However, they are also even easier to cure than mild traumas, which means that keeping these will usually mean keeping a mild trauma along with it.
add: Mobs can only naturally have one mild trauma and one severe or special trauma.
balance: Brain damage will now kill and ruin the brain if it goes above 200. If it somehow goes above 400, the brain will melt and be destroyed completely.
balance: Many brain-damaging effects have been given a damage cap, making them non-lethal.
del: The Unintelligible mutation has been removed and made into a brain trauma.
del: Brain damage no longer makes using machines a living hell.
add: Abductors give minor traumas to people they experiment on.
/:cl:

WIP because i need to flesh out more traumas in the list, but the mechanics are in place.

As always i'm open to suggestions for changes and, obviously, more trauma ideas.

Closes #27330

XDTM committed:
Reworks Brain Damage

XDTM committed:
mechanics

XDTM committed:
rebalancing

XDTM committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into RealBrainDamage

# Conflicts:
#	code/modules/mob/living/carbon/examine.dm

kevinz000 commented on code/datums/brain_damage/special.dm:
```diff
@@ -0,0 +1,150 @@
+#define OWNER 0
+#define STRANGER 1
+
+/datum/brain_trauma/special/split_personality
```
This seems like it's going to get very, VERY griefy.

XDTM replied:
may have to add some message to say that the ghost has to care for its own body and shit

Iamgoofball replied:
this is hillarious and I love it, it'll be fine, just add an admin log when people switch and make sure the ckeys are updated

ChangelingRain commented on code/datums/brain_damage/severe.dm:
```diff
@@ -0,0 +1,62 @@
+	lose_text = "<span class='notice'>You can feel your limbs again!</span>"
+
+/datum/brain_trauma/severe/paralysis/on_life()
+	owner.Knockdown(200)
```
hulk prevents?

XDTM replied:
oh right

XDTM committed:
hulks in wheelchairs

Anonmare commented:
Suggested traumas:-

Minor Trauma:
Phantom illness
Patient believes they are suffering from an illness and will manifest the symptoms of the disease they believe they have, the patient cannot be convinced that they are not sick unless they are treated.
Pick a random, non-special, non-lethal disease. The patient will suffer the symptoms of that disease, will see themselves as having it when examining themselves with an analyzer or medHUD.

Severe Trauma:
Cerebeal Haemorrhaging 
Patient is bleeding into their brain from a severe trauma and steadily losing blood. Patient will die if left untreated, saline-glucose and oxygen treatments can staunch the damage but death is assured without treatment. 
Patient suffers a constant and moderate amount of bloodloss and will need surgery to have it corrected, species that do not have blood will suffer no ill effects from this trauma. 

Special Trauma: 
Secret Agent
Patient believes that they are an undercover spy for Nanotrasen and will pursue their objective to further the corporation's ends.
Makes the patient into an abductee-style antagonist with a bizarre, non-sensical objective. As a pseudo antag, they are not allowed to kill unless in self-defence (no pre-emptive self-defence does not count).

DoctoreMad commented:
Thinkingaboutit, wouldn't it be better to make a thread on the TGstation forums?

XDTM commented:
Probably, but i don't want to rewrite all the features again in a post right now

factoryman942 commented:
>balance: Brain damage will now kill if it goes above 200.
it's time for #30161

RandomMarine commented:
I'm expecting people suffering from a split personality to get killed by their own metafriends because they think they're a ling that took their identity.

Armhulen commented:
split personality sounds like some extremely stupid fun make it an admin smite

Armhulen commented:
also is this testmergable

XDTM commented:
As i mentioned on github, the mechanics should be working, but there aren't many traumas, which means that, for example, paralysis will happen more often than i intended, and special will always be split personality (which i totally want to see ingame tho)

JJRcop commented:
I've wondered about tying brain damage to the brain organ's obj damage, might be neat to look into.

XDTM commented:
shouldn't be too hard, the annoying part would be to replace all the
`guy.traumas`
with
`var/obj/item/brain/B = guy.getorganslot("brain")`
`B.traumas`

Armhulen commented:
/datum/martial_art/psychotic_brawl

soon, lads

JJRcop commented:
By the way I mean like, https://github.com/tgstation/tgstation/blob/fa767ae7cbfea216be090d08b33955f272dbae76/code/game/objects/objs.dm#L11-L12
also [code/game/objects/obj_defense.dm](https://github.com/tgstation/tgstation/blob/fa767ae7cbfea216be090d08b33955f272dbae76/code/game/objects/obj_defense.dm)

zamolxius commented:
absolutely amazing,  thank you

QualityVan commented:
worth mentioning that #30947 recently tried the whole 'enough brain damage kills you' bit and it got closed

XDTM commented:
On this pr, however, most sources got capped to prevent that; you can't just biblebash some guy to braindeath (actually, it will never go beyond a minor trauma). If you want to kill someone via brain damage, chemicals are the way to go.

DaxDupont commented on tgstation.dme:
```diff
@@ -13,7 +13,7 @@
 
 // BEGIN_INCLUDE
-#include "_maps\basemap.dm"
+#include "_maps\runtimestation.dm"
```
10/10

XDTM committed:
yup

XDTM committed:
bugsquash

Supermichael777 commented:
Severe trauma: Prosopagnosia
Victim sees everyone as unknown regardless of whether or not their face is concealed.

Anonmare commented:
**Severe Traumas**

Aphasia:
Patient is unable to speak coherently and their words will come out completely different than what they typed, either using the wrong words, putting them in different order etc.

Language discord:
Randomly gain and/or lose a language known. Patients who lose Common can no longer understand other crewmembers and may ave difficulty expressing to others what is wrong. It is possible, however, to gain languages you didn't know and be able to communicate with other races you couldn't before. 

Left/Right side paralysis:
Severe trauma to the brain has resulted in complete paralysis of one side of the patient's body. Patients are unable to use their affected side's arm and their movement is decreased (effectively they will act as though their two limbs have been amputated)

**Special Traumas**

Space Madness:
Patient becomes incredibly paranoid and sees enemies everywhere and unable to exercise good judgement over what is real and what is not. They are extremely violent and possessed of inhuman durability and will require heavy sedation to incapacitate. 
Effects: Patient becomes incredibly stun resilient and is no longer affected by slowdown and will randomly lose control of their character for brief periods, whereupon they will assault the nearest living thing. They also see everyone as some form of antagonist (traitors with pistols/e-bows/e-swords, Nuke OPs, Wizards, Changelings etc.) and hallucinate that people are attacking them even when they are not.

Psychic Unlock:
By random chance, the trauma the patient has received has caused them to gain psychic powers! The odds of this happening are incredibly rare, make sure to inform Nanotrasen of this and submit the crewmember to further _study_.
Effects: Patient gains the following abilities: Whisper (One-way telepathic messages to someone in their line of sight), Psychic scream (Behaves the same way as the Changeling ability), Telekinesis (as the genetics power), Blind (as the Wizard spell) and Astral Projection (Temporarily leave your body and become an incorporeal, ghostly duplicate of yourself and leaving your body comatose for the duration. Incorporeal duplicates can pass through walls and see Revenants/ghosts but cannot hear them, they also cannot use machinery or pick-up items and will die in one hit, sending them back to their bodies but cause the body to throw up and suffer stamina damage.)

XDTM committed:
Adds narcolepsy, phobias and agnosia. Not guaranteed 100% bug-free yet, testmerge at your own risk.

XDTM commented:
Narcolepsy causes you to fall asleep randomly.
Agnosia causes the victum to see everyone else as faceless people, with their names unknown both when examines, interacted with, or spoken from.
Phobias causes you to be afraid of a certain random concept, and seeing certain mobs, objects, turfs, or hearing certain words related to that concept will cause a random fear reaction.

XDTM committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into RealBrainDamage

# Conflicts:
#	code/modules/mob/living/carbon/human/species.dm
#	tgstation.dme

Iamgoofball commented:
Make sure to remove bible bashing brain damage, this became an issue with Goon Medical when I ported it awhile back.

XDTM committed:
lizard phobia

Iamgoofball commented on code/datums/brain_damage/mild.dm:
```diff
@@ -0,0 +1,231 @@
+	owner.dna.remove_mutation(UNINTELLIGIBLE)
+	..()
+
+/datum/brain_trauma/mild/concussion
```
We could probably make this occur to heavy knocks to the head instead of the instant stun.

Iamgoofball commented on code/datums/brain_damage/mild.dm:
```diff
@@ -0,0 +1,231 @@
+/datum/brain_trauma/mild/phobia/New(mob/living/carbon/C, _permanent, specific_type)
+	phobia_type = specific_type
+	if(!phobia_type)
+		phobia_type = pick("spiders", "space", "security", "clowns", "assistants")
```
no skeletons? in my october?

0/10

Iamgoofball replied:
fixed this here
https://github.com/XDTM/tgstation/pull/4

Iamgoofball commented on code/datums/brain_damage/severe.dm:
```diff
@@ -0,0 +1,108 @@
+	owner.Knockdown(200, ignore_canknockdown = TRUE)
+	..()
+
+/datum/brain_trauma/severe/blindness/on_lose()
```
this needs to be paralysis, not blindness

Iamgoofball replied:
To repeat you pathed this as blindness

Iamgoofball commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -146,14 +146,21 @@ MASS SPECTROMETER
+	else if (M.getBrainLoss() >= 120)
+		to_chat(user, "\t<span class='alert'>Severe brain damage detected. Subject likely to have mental traumas.</span>")
+	else if (M.getBrainLoss() >= 45)
 		to_chat(user, "\t<span class='alert'>Brain damage detected. Subject may have had a concussion.</span>")
```
change this line since concussions are an actual thing

Iamgoofball commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -425,3 +428,19 @@
+	for(var/datum/brain_trauma/BT in traumas)
+		BT.on_life()
+
+	if(brainloss >= BRAIN_DAMAGE_DEATH) //rip
```
People are going to whine about this a lot, and they will go out of their way to grief with this to declare that we should remove it. See past this. This is a good feature for the overhaul.

XDTM replied:
I've already prevented most griefing by capping most sources of brain damage. Chems are the most surefire way of brain death, and that takes effort.

Iamgoofball commented on code/modules/mob/living/carbon/say.dm:
```diff
@@ -1,4 +1,6 @@
 /mob/living/carbon/treat_message(message)
+	for(var/datum/brain_trauma/trauma in traumas)
```
typeless loop

Iamgoofball commented on code/modules/surgery/brain_surgery.dm:
```diff
@@ -0,0 +1,41 @@
+/datum/surgery/brain_surgery
+	name = "brain surgery"
```
rename to lobotomy

JJRcop replied:
if the subject is a plantperson it should be lobotany

Iamgoofball replied:
ahahahaha

XDTM replied:
i'm torn on this, with brain surgery we'd get all the "it's not brain surgery" jokes

Iamgoofball replied:
Hmm

Iamgoofball committed:
Did you know there's a skeleton inside of you RIGHT NOW?

Iamgoofball commented:
https://github.com/XDTM/tgstation/pull/4

XDTM committed:
Fixes for the new stuff

XDTM committed:
The issue with those is that in most rounds you won't see them

I'll keep them available so they can be added manually, for example by events and such

XDTM committed:
Merge pull request #4 from Iamgoofball/patch-318

Did you know there's a skeleton inside of you RIGHT NOW?

XDTM committed:
Goof reviews

XDTM commented:
Added 20% chance of concussion if you suffer a head crit. Which is also the same thing that has a chance of converting revs. Fun!

Commit:
Brainloss is based off brain organ integrity

removes brainloss var

Commit:
Replaces some getorganslot with defines

JJRcop commented:
I made a PR to this branch that removes the brainloss var and instead makes brainloss based off of the brain organ's obj_integrity.

lordpidey commented on code/modules/paperwork/contract.dm:
```diff
@@ -45,11 +45,6 @@
-			var/mob/living/carbon/human/N = M
-			if(!istype(N.head, /obj/item/clothing/head/helmet))
-				N.adjustBrainLoss(10)
-				to_chat(N, "<span class='danger'>You feel dumber.</span>")
```
So part of the reason for this brainloss is to prevent someone from selling their soul and recovering it repeatedly with their own employment contract.  Eventually the brainloss would get so bad that you would be unable to sell your soul.

XDTM replied:
I replaced the brainloss check with a DUMB check anyway.
Maybe the limitation should be a bit more obvious, like "You still remember the terms of your contract! You cannot give away your soul to third parties."

lordpidey replied:
But without the brainloss here, having your soulreturned won't cause the dumb condition, which is the point of the damage here, preventing infinite contract re-signing.

JJRcop replied:
This is a balance change that wasn't addressed

lordpidey commented on code/modules/surgery/brain_surgery.dm:
```diff
@@ -0,0 +1,41 @@
+	var/obj/item/organ/brain/B = target.getorganslot("brain")
+	if(!B)
+		to_chat(user, "It's hard to do surgery on someone's brain when they don't have one.")
+		return FALSE
```
Don't we need a return true here?

XDTM replied:
pretty sure that can_start shouldn't be true without a brain

lordpidey replied:
I meant directly below.  There is no path that returns true for this proc.

lordpidey replied:
So....  this proc can still NEVER return true.

Are you SURE this won't cause an issue?

lordpidey commented on code/modules/surgery/brain_surgery.dm:
```diff
@@ -0,0 +1,41 @@
+		return FALSE
+
+/datum/surgery_step/fix_brain/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	user.visible_message("[user] begins to fix [target]'s eyes.", "<span class='notice'>You begin to fix [target]'s brain...</span>")
```
First string says you're fixing eyes, not brain.

XDTM committed:
eyes

XDTM committed:
Merge pull request #5 from JJRcop/brainloss-brain-organ

Brainloss is on the brain organ

XDTM committed:
put the traumas in the brainnnn

XDTM committed:
text

lordpidey commented on code/datums/brain_damage/special.dm:
```diff
@@ -0,0 +1,150 @@
+
+/mob/living/split_personality/say(message)
+	to_chat(src, "<span class='warning'>You cannot speak, your other self is controlling your body!</span>")
+	return FALSE
```
Perhaps we should let the dormant personality talk to the controller?

For extra hilarity, make the controller have to talk out loud to respond.

XDTM replied:
I pretty much took the code for this from borers, but their saycode was a copypasted horror which i decided to avoid. Also, i kinda like the idea of personalities being separated from each other completely.

PKPenguin321 commented:
>balance: Brain damage will now kill if it goes above 200.
balance: Many brain-damaging effects have been given a damage cap, making them non-lethal.

Any examples of these? What are things like the bible capped at? What sources are still lethal?
Brain damage inducing poisons might be a little bit nuts

ExcessiveUseOfCobblestone commented:
guess you've never heard of amatoxin

Iamgoofball commented:
@pkpenguin12 I'm pretty sure you'd notice the trauma effects

XDTM committed:
args

XDTM committed:
Adds godwoken, puts split personality in severe, adds brainwashing

JJRcop commented:
When you're done with this add "[READY]" to the title and @tgstation-server will automatically remove the Work in Progress tag.

JJRcop commented:
Also maybe mention that 400 brain damage destroys the brain in the changelog.

New trauma:
**Severe**
Dependent: Patient may involuntarily grab someone next to them aggressively.

XDTM committed:
Bluespace prophet special trauma (untested)

XDTM commented:
Added a couple special traumas:

Godwoken Syndrome will rarely make you speak with the voice of god instead of your normal voice (with cooldown to prevent spam powergaming)

Bluespace Prophet will cause the patient to see special portals that only they can cross to reach the other one, and that is invisible and untouchable to everything else. (Lorewise this is the patient "understanding" the flow of bluespace, allowing them to take shortcuts when things align in the right way)

GuyonBroadway commented:
I might make the suggestion that rather than have brain damage flat cause death, rather add some traumas that result in death. 

For example how about unstable pyrokinesis, every so often you catch fire and release a small burst of flame. 

Or how about hallucinations that do REAL damage.

Armhulen commented:
severe trauma that makes you fight your inner demons or something please!

GuyonBroadway commented:
Actually that could make for an interesting truama, it spawns hostile deadly mobs but you can kill them. Kill enough and you purge the trauma.

DoctoreMad commented:
I would complain about making a trauma fixable by means OTHER than brain surgery but I realized that if it wasn't possible, a hallucinated mob would probably try to kill you during your surgery.

lordpidey commented:
Or just make N2O/morphine cause the hallucinations to stop.  Gives people a reason to actually use anesthesia for surgery.

Bawhoppen commented:
@lordpidey We already have a reason to use anesthesia for surgery?
It's the same one as there is in real life.

DoctoreMad commented:
Holy shit people felt real pain when I did surgery on them without anesthesia?

lordpidey commented:
@Bawhoppen, what percentage of surgeries have some form of anesthesia done on TG?  I'd wager less than 10%.  We should be encouraging it.

XDTM commented:
Maybe unconscious targets would have improved success chance, and we could lower the base chance to 80% or so, while adding penalties for failing

lordpidey commented:
Good ideas, @XDTM, but that's for another PR.

Iamgoofball commented:
I'm pretty sure we do improve chances for ghetto surgery if the target is knocked out

lordpidey commented:
I'm not seeing that in the code, @Iamgoofball.  Unless it's tied to a specific surgery.

The only things that matter is tool quality, table quality, and if the patient has been sterilized.

Iamgoofball commented:
Hm

We should implement that

DoctoreMad commented:
That's what we were dwelling on a minute ago.

Bawhoppen commented:
Not my fault that so few people RP

XDTM committed:
Some minor stuff

XDTM commented:
Updates!

- Godwoken syndrome now shows as "god delusion" to medical scanners
- Permanent traumas no longer prevent new traumas of the same type
- Abductors now cause minor traumas on people they experiment on

swindly commented on code/datums/brain_damage/mild.dm:
```diff
@@ -0,0 +1,250 @@
+
+/datum/brain_trauma/mild/phobia
+	name = "Phobia"
+	desc = "Patient is unreasonaly afraid of something."
```
unreasonably

Ratvar commented:
Can ashwalkers operate other ashwalkers to remove Unintelligible?

XDTM commented:
Technically yes, but failing the surgery deals a lot of brain damage

Slignerd commented on code/datums/brain_damage/mild.dm:
```diff
@@ -0,0 +1,250 @@
+			trigger_mobs = list(/mob/living/simple_animal/hostile/lizard) //they're hostile! of course they're scary!
+			trigger_species = list(/datum/species/lizard)
+		if("spooky skeletons")
+			trigger_words = list("skeleton", "rattle me bones", "milk", "xylophone", "bone", "calcium", "i want to get off mr bones wild ride", "the ride never ends")
```
Should just shorten it to "mr bones wild ride".

Cyberboss replied:
What @Slignerd said

YPOQ commented:
#27330

XDTM committed:
Monophobia, and tweaks

XDTM committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into RealBrainDamage

# Conflicts:
#	code/modules/mob/living/carbon/carbon.dm
#	icons/mob/simple_human.dmi

XDTM committed:
More fixes

XDTM commented:
Monophobia will make the mob feel sick and dizzy if they're alone (aka not near a player-controlled mob or a pet). If they stay alone for too long it gets more dangerous, with the stress causing blood vomiting, hallucinations, and possibly a heart attack. (Making this the first really lethal brain trauma)

XDTM committed:
fix

JJRcop commented on code/datums/brain_damage/severe.dm:
```diff
@@ -0,0 +1,193 @@
+	scan_desc = "severe monophobia"
+	gain_text = ""
+	lose_text = "<span class='notice'>You feel like you could be safe on your own.</span>"
+	var/stress = 0
```
No need to change anything, but this is a likely candidate for statusification in the future.

lordpidey commented:
How about you add an illiterate trauma.  Take a look at the `is_literate()` proc.

PKPenguin321 commented on code/datums/brain_damage/severe.dm:
```diff
@@ -0,0 +1,193 @@
+	..()
+
+/datum/brain_trauma/severe/mute/on_lose()
+	owner.disabilities &= ~MUTE
```
What if you gain mute from another source like genetics then this happens

lordpidey replied:
How about you route the ability to hear messages through the brain?  This would also allow for the ability for a brain trauma that sutbtly changes messages.  Such as one that replaces locations with other locations.

PKPenguin321 replied:
Would require a bit of refactoring, but you could do disabilities like martial arts where there are regular ones and temporary ones that take priority, allowing you to add and remove temporary ones without affecting normal ones

XDTM replied:
now if only someone would merge that

Iamgoofball commented:
Guyon, get good and fugg off.

GuyonBroadway commented:
**FINISH :clap: G.O.A.P :clap: AI**

Iamgoofball commented:
Remie will whine and I don't want to deal with her not understanding what agpl is

DoctoreMad commented:
>I don't want to deal with X's whining about Y. That's why I dropped the feature. Believe me.

Can you even do anything right

XDTM committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into RealBrainDamage

# Conflicts:
#	icons/effects/effects.dmi

XDTM committed:
a

XDTM committed:
new tail

RemieRichards commented on code/datums/brain_damage/brain_trauma.dm:
```diff
@@ -0,0 +1,49 @@
+//Brain Traumas are the new actual brain damage. Brain damage itself acts as a way to acquire traumas: every time brain damage is dealt, there's a chance of receiving a trauma.
+//This chance gets higher the higher the mob's brainloss is. Removing traumas is a separate thing from removing brain damage: you can get restored to full brain operativity,
+//but keep the quirks, until repaired by mannitol (for mild/special ones) or brain surgery (for severe ones).
+/datum/brain_trauma
```
any reason these aren't an extension of status effects?

XDTM replied:
They're very different, traumas are permanent until cured, so they don't really need the status effect timer system, react to some mob actions (when speaking, when hearing something, etc.), have to be transferrable through the brain when it's removed from the body, have to be transferred when monkeying/unmonkeying, and some other stuff i can't remember at the moment.

ninjanomnom replied:
This is more of an argument to be made for status effects to move to components tbh

XDTM committed:
Discoordination and Muscle Weakness

XDTM commented:
New traumas!
Muscle Weakness (Mild) causes the patient to occasionally fall, drop held items, or feel intense weakness.
Discoordination (Severe) prevents the patient from using complex tools or machinery, much like a monkey.

XDTM committed:
Merge branch 'master' into RealBrainDamage

XDTM committed:
Merge branch 'master' into RealBrainDamage

lordpidey commented on code/modules/mob/living/carbon/human/human_helpers.dm:
```diff
@@ -103,7 +103,9 @@
-	return 1//Humans can use guns and such
+	if(disabilities & MONKEYLIKE)
+		return FALSE
+	return TRUE//Humans can use guns and such
```
return (disabilities & MONKEYLIKE)

XDTM replied:
i think that'd be less reader-friendly, but if it's significant in performance i'll do that

lordpidey replied:
I'm not sure how faster it is, but if(X) return true, else return false just rubs me the wrong way.

ninjanomnom replied:
I'd prefer we not use the `return (disabilities & MONKEYLIKE)` style as it inhibits future expansion of the proc

optimumtact commented:
given that no maintainer has commented on this (that I can see at least) I'm closing it in 24 hours unless one of them steps in to make this ready

@tgstation/commit-access

XDTM commented:
ree

DoctoreMad commented:
@optimumtact You just did.
I thought they didn't because they were working with the R&D changes?

Cyberboss commented on code/datums/brain_damage/special.dm:
```diff
@@ -0,0 +1,88 @@
+		first.seer = owner
+		second.seer = owner
+
+/obj/effect/hallucination/simple/bluespace_stream
```
should be in the hallucinations module

XDTM replied:
you sure? the hallucination module is pretty much organized with actual hallucinations grouped with necessary resources, i'm not sure if this would be any more appropriate there

Cyberboss commented on code/datums/brain_damage/split_personality.dm:
```diff
@@ -0,0 +1,218 @@
+	owner_backseat = new(owner, src)
+
+/datum/brain_trauma/severe/split_personality/proc/get_ghost()
+	set waitfor = 0
```
FALSE

Cyberboss commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -1,13 +1,19 @@
 /mob/living/carbon/human/examine(mob/user)
-//this is very slightly better than it was because you can use it more places. still can't do \his[src] though.
+
+	if(user.disabilities & AGNOSIA && user != src)
```
parenthisis required

Cyberboss commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -28,6 +28,13 @@
 	. = ..()
 
+	//Image override for agnosiacs
+	var/image/I = image(icon = 'icons/mob/simple_human.dmi', icon_state = "faceless", loc = src)
```
mutable_appearance

Cyberboss commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -437,3 +440,19 @@
+////////////////
+
+/mob/living/carbon/proc/handle_brain_damage()
+	for(var/datum/brain_trauma/BT in get_traumas())
```
typeless loop

Cyberboss commented on code/modules/mob/transform_procs.dm:
```diff
@@ -79,6 +79,8 @@
 		O.setBrainLoss(getBrainLoss(), 0)
 		O.updatehealth()
 		O.radiation = radiation
+		for(var/datum/brain_trauma/BT in get_traumas())
```
typeless

Cyberboss commented on code/modules/mob/transform_procs.dm:
```diff
@@ -236,6 +238,8 @@
 		O.setBrainLoss(getBrainLoss(), 0)
 		O.updatehealth()
 		O.radiation = radiation
+		for(var/datum/brain_trauma/BT in get_traumas())
```
typeless

Cyberboss commented on code/modules/surgery/brain_surgery.dm:
```diff
@@ -0,0 +1,41 @@
+/datum/surgery/brain_surgery/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/brain/B = target.getorganslot(ORGAN_SLOT_BRAIN)
+	if(!B)
+		to_chat(user, "It's hard to do surgery on someone's brain when they don't have one.")
```
warning span

XDTM committed:
thanks for the review!

DoctoreMad commented:
_Brain damage station soon_

lordpidey commented on code/datums/brain_damage/mild.dm:
```diff
@@ -162,7 +162,7 @@
 			trigger_species = list(/datum/species/lizard)
 		if("spooky skeletons")
-			trigger_words = list("skeleton", "rattle me bones", "milk", "xylophone", "bone", "calcium", "i want to get off mr bones wild ride", "the ride never ends")
+			trigger_words = list("skeleton", "rattle me bones", "milk", "xylophone", "bone", "calcium", "mr bones wild ride", "the ride never ends")
```
doesn't the word "bone" make "rattle me bones" and "mr bones wild ride" redundant?

XDTM replied:
True, i gotta review goofcode before merging it

XDTM committed:
[B]oneless

JJRcop commented:
I think `"wild ride"` should be put back in, to replace the redundant `"mr bones wild ride"`.

XDTM commented:
Wild ride by itself is not very skeleton-y tho, it's the mr bones part that is spooky and that part contains the word bone

DoctoreMad commented:
@tgstation/commit-access M-merge when

Armhulen commented:
if this stays in limbo i'll finish brawl

DoctoreMad commented:
Finish what?

XDTM commented:
Arm was making an insanity martial art to give as a special brain trauma

XDTM committed:
agnosia fix

JJRcop commented on code/_onclick/item_attack.dm:
```diff
@@ -129,8 +129,12 @@
 	if(hit_area)
 		message_hit_area = " in the [hit_area]"
 	var/attack_message = "[src] has been [message_verb][message_hit_area] with [I]."
+	var/agnosia_message = "Someone has been [message_verb][message_hit_area] with [I]."
```
It's a little weird to have a message in item attack code for one trauma

JJRcop commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -5,6 +5,7 @@
 	icon = 'icons/mob/human.dmi'
 	icon_state = "caucasian_m"
 	appearance_flags = KEEP_TOGETHER|TILE_BOUND|PIXEL_SCALE
+	var/has_agnosia_image = TRUE
```
This is also weird

XDTM replied:
To avoid dummies having it, although it's not strictly necessary

lordpidey replied:
Just have dummies override `generate_agnosia_image()` with a proc that does nothing.

JJRcop commented on code/modules/mob/mob.dm:
```diff
@@ -101,8 +101,10 @@
+// agnosia_self_message is the same, for the self_message
 
-/atom/proc/visible_message(message, self_message, blind_message, vision_distance, ignored_mob)
+/atom/proc/visible_message(message, self_message, blind_message, vision_distance, ignored_mob, agnosia_message, agnosia_self_message)
```
This is quite weird. This is a very well used proc, and two new arguments for a trauma is undesirable. Unfortunately I don't think there is better way to achieve this otherwise without dynamically changing the names of the subjects dependent of recipient.

AnturK replied:
I have to agree with these, these are horribly intrusive and these systems shouldn't need to know about something as specific as brain traumas to work properly.

JJRcop reviewed (commented):
Agnosia is too far reaching into places it shouldn't be, instead of being more modular. I think that, while agnosia is a really cool trauma, it's way too snowflake, and I can't think of a proper way to do it unless we have some refactors to the visible message chain and how mobs see each other. I think it should be removed :slightly_frowning_face:

XDTM commented:
Yeah, it started out as a visual hallucinations then it spiraled down into the mess it is, but i grew attached to it since it actually works now

If it's a dealbreaker i'll cut it out, but i'll be sad

XDTM committed:
.

XDTM commented:
I'm gonna strip out agnosia, i'm not sure how i got so delusional about this being ok

XDTM committed:
go away agnosia

XDTM committed:
pretend it didn't happen

JJRcop commented:
RIP agnosia, we can't handle you rigt now :frowning_face:

XDTM committed:
Merge branch 'master' into RealBrainDamage

XDTM committed:
Merge branch 'master' into RealBrainDamage

DoctoreMad commented:
>[Shitcode-Free]

Check failed.
Also can anyone figure out a simpler way to add Agnosia? It was pretty unique to me.

XDTM committed:
Shitcode-free-er

Cyberboss commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -48,7 +48,7 @@
 	if(istype(M) && on && user.zone_selected in list("eyes", "mouth"))
 
-		if((user.disabilities & CLUMSY || user.getBrainLoss() >= 60) && prob(50))	//too dumb to use flashlight properly
+		if((user.disabilities & CLUMSY || user.disabilities & DUMB) && prob(50))	//too dumb to use flashlight properly
```
need parenthesis around the bitwise ops

Cyberboss commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -87,7 +87,7 @@ MASS SPECTROMETER
 
 	// Clumsiness/brain damage check
-	if ((user.disabilities & CLUMSY || user.getBrainLoss() >= 60) && prob(50))
+	if ((user.disabilities & DUMB || user.disabilities & CLUMSY) && prob(50))
```
ditto

Cyberboss commented on code/modules/assembly/mousetrap.dm:
```diff
@@ -20,7 +20,7 @@
 			if(ishuman(usr))
 				var/mob/living/carbon/human/user = usr
-				if((user.getBrainLoss() >= 60) || user.disabilities & CLUMSY && prob(50))
+				if((user.disabilities & DUMB || user.disabilities & CLUMSY) && prob(50))
```
ditto

Cyberboss commented on code/modules/assembly/mousetrap.dm:
```diff
@@ -76,7 +76,7 @@
 		to_chat(user, "<span class='notice'>You arm [src].</span>")
 	else
-		if(((user.getBrainLoss() >= 60) || user.disabilities & CLUMSY) && prob(50))
+		if(((user.disabilities & DUMB) || user.disabilities & CLUMSY) && prob(50))
```
ditto

Cyberboss commented on code/modules/assembly/mousetrap.dm:
```diff
@@ -92,7 +92,7 @@
 /obj/item/device/assembly/mousetrap/attack_hand(mob/living/carbon/human/user)
 	if(armed)
-		if(((user.getBrainLoss() >= 60) || user.disabilities & CLUMSY) && prob(50))
+		if(((user.disabilities & DUMB) || user.disabilities & CLUMSY) && prob(50))
```
ditto

XDTM committed:
bitwisdom

XDTM committed:
Merge branch 'RealBrainDamage' of https://github.com/XDTM/tgstation into RealBrainDamage

XDTM committed:
psychotic brawling

XDTM commented:
Added psychotic brawling, a new special trauma that makes you fight unpredictably! Hug your enemies, then kick them across the room.

Sorry @Armhulen, it was taking a while for you to finish it

XDTM committed:
i guess this might be important

lordpidey commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -48,7 +48,7 @@
 	if(istype(M) && on && user.zone_selected in list("eyes", "mouth"))
 
-		if((user.disabilities & CLUMSY || user.disabilities & DUMB) && prob(50))	//too dumb to use flashlight properly
+		if(((user.disabilities & CLUMSY) || (user.disabilities & DUMB)) && prob(50))	//too dumb to use flashlight properly
```
This can be simplified to `(user.disabilities & (CLUMSY | DUMB))`

lordpidey commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -87,7 +87,7 @@ MASS SPECTROMETER
 
 	// Clumsiness/brain damage check
-	if ((user.disabilities & DUMB || user.disabilities & CLUMSY) && prob(50))
+	if (((user.disabilities & DUMB) || (user.disabilities & CLUMSY)) && prob(50))
```
Same

lordpidey commented on code/modules/assembly/mousetrap.dm:
```diff
@@ -20,7 +20,7 @@
 			if(ishuman(usr))
 				var/mob/living/carbon/human/user = usr
-				if((user.disabilities & DUMB || user.disabilities & CLUMSY) && prob(50))
+				if(((user.disabilities & DUMB) || (user.disabilities & CLUMSY)) && prob(50))
```
Same

lordpidey commented on code/modules/assembly/mousetrap.dm:
```diff
@@ -76,7 +76,7 @@
 		to_chat(user, "<span class='notice'>You arm [src].</span>")
 	else
-		if(((user.disabilities & DUMB) || user.disabilities & CLUMSY) && prob(50))
+		if(((user.disabilities & DUMB) || (user.disabilities & CLUMSY)) && prob(50))
```
Same

lordpidey commented on code/modules/assembly/mousetrap.dm:
```diff
@@ -92,7 +92,7 @@
 /obj/item/device/assembly/mousetrap/attack_hand(mob/living/carbon/human/user)
 	if(armed)
-		if(((user.disabilities & DUMB) || user.disabilities & CLUMSY) && prob(50))
+		if(((user.disabilities & DUMB) || (user.disabilities & CLUMSY)) && prob(50))
```
Same

lordpidey reviewed (commented):
Bitwise operators can be further improved.

ninjanomnom commented on code/datums/brain_damage/mild.dm:
```diff
@@ -0,0 +1,278 @@
+	owner.stuttering = 0
+	..()
+
+#define BRAIN_DAMAGE_FILE "brain_damage_lines.json"
```
This is already defined elsewhere

XDTM replied:
fixed

ninjanomnom replied:
The value is used elsewhere, just put the define in one of the defines files

XDTM replied:
Why is it even a define? All it saves is writing a .json at the end, while clouding which file it actually is to code readers

ninjanomnom replied:
because the exact location and name of the file can more easily be changed from one location then

ninjanomnom commented on code/datums/brain_damage/mild.dm:
```diff
@@ -0,0 +1,278 @@
+	trigger_turfs = typecacheof(trigger_turfs)
+	trigger_mobs = typecacheof(trigger_mobs)
+	trigger_objs = typecacheof(trigger_objs)
+	trigger_species = typecacheof(trigger_species)
```
These typecaches should be static somewhere. Either static in here or if they're used elsewhere a global list

XDTM replied:
unless i make a separate global list for every trigger type for every phobia i can't do that, i think, and i'd rather not clutter global vars needlessly

ninjanomnom replied:
It's not needless, it's better than the performance cost here and you could just put it in a subsystem

ninjanomnom replied:
Or for that matter a list of lists

ninjanomnom replied:
@XDTM you still need to deal with this

ninjanomnom commented on code/datums/brain_damage/split_personality.dm:
```diff
@@ -0,0 +1,218 @@
+	owner_backseat = new(owner, src)
+
+/datum/brain_trauma/severe/split_personality/brainwashing/get_ghost()
+	set waitfor = 0
```
FALSE

ninjanomnom commented on code/modules/mob/living/damage_procs.dm:
```diff
@@ -219,17 +219,11 @@
 
 /mob/living/proc/getBrainLoss()
-	return brainloss
+	. = 0
```
always 0?

JJRcop replied:
Only carbons can have brains.

ninjanomnom replied:
ah it's overridden, okay

XDTM committed:
latest reviews

XDTM committed:
.

JJRcop commented:
getting stale

XDTM committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into RealBrainDamage

# Conflicts:
#	code/datums/mutations.dm
#	code/game/alternate_appearance.dm
#	icons/effects/effects.dmi

XDTM committed:
conflict-free

Cyberboss commented:
Not hearing any deaththroes from sybil so do what @ninjanomnom wants and this is good to go

XDTM committed:
addresses it

XDTM committed:
surgery fix

XDTM commented:
@ninjanomnom is this fine?

ninjanomnom commented on code/datums/brain_damage/phobia.dm:
```diff
@@ -0,0 +1,162 @@
+GLOBAL_LIST_INIT(phobia_skeleton_words, list("skeleton", "milk", "xylophone", "bone", "calcium", "the ride never ends"))
+GLOBAL_LIST_INIT(phobia_skeleton_objs, typecacheof(/obj/item/organ/tongue/bone, /obj/item/clothing/suit/armor/bone, /obj/item/stack/sheet/bone,\
+				/obj/item/reagent_containers/food/snacks/meat/slab/human/mutant/skeleton, /obj/effect/decal/remains/human))
+GLOBAL_LIST_INIT(phobia_skeleton_species, typecacheof(/datum/species/skeleton, /datum/species/plasmaman))
```
make these a list of lists organized so you access like `GLOB.phobia_words["spider"]`

Also these being in a subsystem would probably be preferable

ninjanomnom commented on code/datums/brain_damage/phobia.dm:
```diff
@@ -0,0 +1,162 @@
+	scan_desc += " of [phobia_type]"
+	switch(phobia_type)
+		if("spiders")
+			trigger_words = GLOB.phobia_spider_words
```
then you can just do `trigger_words = GLOB.phobia_words[phobia_type]`

XDTM committed:
a masterpiece

XDTM committed:
fix

XDTM commented:
@ninjanomnom Done!

XDTM committed:
fix again

Buggy123 commented:
Haven't see anyone ask this - will cloned people require surgery? They _do_ start with brain damage, after all.

XDTM commented:
They aren't able to reach the amount of brain damage required for heavy traumas by cloning alone, so mannitol is still the go-to after cloning.

ninjanomnom commented on code/controllers/subsystem/traumas.dm:
```diff
@@ -0,0 +1,40 @@
+								"skeletons" = list("skeleton", "milk", "xylophone", "bone", "calcium", "the ride never ends")
+								)
+
+	var/list/phobia_mobs = list("spiders"  = typecacheof(/mob/living/simple_animal/hostile/poison/giant_spider),
```
I'd prefer these lists be created in initialize.

XDTM replied:
i could've sworn i did that, maybe i forgot to push it

ninjanomnom commented on code/controllers/subsystem/traumas.dm:
```diff
@@ -0,0 +1,48 @@
+						"skeletons" = list("skeleton", "milk", "xylophone", "bone", "calcium", "the ride never ends")
+					   )
+
+	phobia_mobs = list("spiders"  = typecacheof(/mob/living/simple_animal/hostile/poison/giant_spider),
```
I'd prefer these were created in initialize

ninjanomnom commented on code/controllers/subsystem/traumas.dm:
```diff
@@ -0,0 +1,40 @@
+								"skeletons" = list("skeleton", "milk", "xylophone", "bone", "calcium", "the ride never ends")
+								)
+
+	var/list/phobia_mobs = list("spiders"  = typecacheof(/mob/living/simple_animal/hostile/poison/giant_spider),
```
you did I was looking at old code

XDTM committed:
config

XDTM commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -143,3 +176,58 @@
+		return
+
+	if(cure_permanent || !trauma.permanent)
+		qdel(trauma)
```
i suppose that proc is pointless, yeah.

XDTM commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -143,3 +176,58 @@
+	for(var/X in traumas)
+		if(ignore_thresholds || (istype(X, BRAIN_TRAUMA_MILD) && brainloss < BRAIN_DAMAGE_MILD))
+			cure_trauma(X, cure_permanent)
+		else if(ignore_thresholds || ((istype(X, BRAIN_TRAUMA_SEVERE) || istype(X, BRAIN_TRAUMA_SPECIAL)) && brainloss < BRAIN_DAMAGE_SEVERE))
```
i'm removing thresholds anyway, there's no real reason to have them

XDTM commented on code/modules/mob/living/carbon/damage_procs.dm:
```diff
@@ -193,3 +193,37 @@
+	B.adjust_brain_damage(amount, maximum)
+	if(amount <= 0) //cut this early
+		return
+	var/brainloss = getBrainLoss()
```
if i do that i can no longer check if it was a heal and cut the rest of the proc, since it currently returns the amount adjusted.

XDTM committed:
damage formula reworked

XDTM commented:
reworked the brain damage formula to scale more linearly. The logic still is that more hit damage = more chance and more existing damage = more chance.

XDTM committed:
who needs to test anyway

XDTM commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -143,3 +176,58 @@
+
+
+//Add a specific trauma
+/obj/item/organ/brain/proc/gain_trauma(datum/brain_trauma/trauma, permanent = FALSE, ...)
```
done

QualityVan commented on code/datums/brain_damage/severe.dm:
```diff
@@ -0,0 +1,185 @@
+				to_chat(owner, "<span class='warning'>You feel sick...</span>")
+			else
+				to_chat(owner, "<span class='warning'>You feel really sick at the thought of being alone!</span>")
+			addtimer(CALLBACK(owner, /mob/living/carbon.proc/vomit, high_stress ? FALSE : TRUE), 50) //blood vomit if high stress
```
Some reason this shouldn't just be !high_stress ?

QualityVan commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -133,6 +145,27 @@
+		if(adjusted_amount >= 0.1)
+			take_damage(adjusted_amount)
+		else if(adjusted_amount <= -0.1)
+			obj_integrity = min(max_integrity, obj_integrity-adjusted_amount)
```
So taking brain damage from a source with a lower cap than your current brain damage will... repair brain integrity? And sources with lower maximums will be better at it?

XDTM replied:
goddammit

JJRcop replied:
In testing I don't remember this happening, try it please.

XDTM committed:
fixes some bugs

XDTM committed:
fix 2

XDTM committed:
proper spookin

XDTM committed:
bluespace prophet fixes and improvements

XDTM committed:
no bible healing

XDTM committed:
.

XDTM committed:
normalizes monophobia chances

XDTM committed:
Merge branch 'master' into RealBrainDamage

XDTM committed:
Merge branch 'master' into RealBrainDamage

XDTM committed:
Merge branch 'master' into RealBrainDamage

XDTM committed:
Merge branch 'master' into RealBrainDamage

Cyberboss commented:
@ninjanomnom pls

Cyberboss commented:
@XDTM one more confritur and ill merge

XDTM committed:
Merge branch 'master' into RealBrainDamage

XDTM commented:
@Cyberboss
